tequilapi/endpoints/location.go

```diff
@@ -0,0 +1,52 @@
+package endpoints
+
+import (
+	"github.com/julienschmidt/httprouter"
+	"github.com/mysterium/node/location"
+	"github.com/mysterium/node/tequilapi/utils"
+	"net/http"
+)
+
+// LocationEndpoint struct represents /location resource and it's subresources
+type LocationEndpoint struct {
+	locationDetector location.Detector
+	locationCache location.Cache
+}
+
+// NewLocationEndpoint creates and returns location endpoint
+func NewLocationEndpoint(locationDetector location.Detector, locationCache location.Cache) *LocationEndpoint {
+	return &LocationEndpoint{
+		locationDetector: locationDetector,
+		locationCache: locationCache,
+	}
+}
+
+// GetLocation responds with original and current countries
+func (ce *LocationEndpoint) GetLocation(writer http.ResponseWriter, request *http.Request, params httprouter.Params) {
+	originalLocation, err := ce.locationCache.Get()
+	if err != nil {
+		utils.SendError(writer, err, http.StatusServiceUnavailable)
+		return
+	}
+
+	currentLocation, err := ce.locationDetector.DetectLocation()
+	if err != nil {
+		utils.SendError(writer, err, http.StatusServiceUnavailable)
+		return
+	}
+
+	response := struct {
+		Original location.Location `json:"original"`
```

Comment:
`original` looks confusing - it doesn't specify what this location actually means. Maybe `startup` would be more clear?

Reply:
`original` is just perfect

Reply:
ok, when you put it like this, it does make sense - but at first look, it didn't